File: Graph Algorithms in Genome Sequencing/Genome seq/Construct the Overlap Graph of a Collection of k-mers/kmer_overlap_graph.cpp
#include <iostream>
#include <vector>
#include <string>

using std::cout;
using std::cin;
using std::endl;
using std::string;
using std::vector;


/*
 * Creates an overlap graph from a list of k-mers
 *
 * \param patterns      a vector of string k-mers
 *
 * \return a string containing an adjacency list representation of the
 * overlap graph as described in the problem specification
 */
string overlap_graph(const vector<string>& patterns) {
    // TODO: your code here
    return "";
}


int main(int argc, char* argv[]) {
    vector<string> patterns;

    string pattern;
    while (cin >> pattern) {
        patterns.push_back(pattern);
    }

    cout << overlap_graph(patterns) << endl;
}

File: Graph Algorithms in Genome Sequencing/Genome seq/Construct the Overlap Graph of a Collection of k-mers/KmerOverlapGraph.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class KmerOverlapGraph {
    /*
     * Creates an overlap graph from a list of k-mers
     *
     * @param patterns      a list of String k-mers
     *
     * @return a string containing an adjacency list representation of the
     * overlap graph as described in the problem specification
     */
    public static String overlapGraph(List<String> patterns) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        List<String> patterns = new ArrayList<String>();

        while (stdin.hasNextLine()) patterns.add(stdin.nextLine());

        System.out.println(overlapGraph(patterns));
    }
}
